src/game_objects/game_world_entity.py

```diff
@@ -57,59 +57,57 @@ def update_movement_velocity(self, game_state, game_world_state):
     def update_hitbox(self):
         self.state[const.HITBOX] = self.state[const.HITBOX_CONFIG][self.state[const.STATE]]
 
-    def apply_collision_detection(self, game_state):
-        new_hitbox_data = self.state[const.HITBOX_CONFIG].get(self.state[const.STATE])
-        hitbox_pos = new_hitbox_data[0]
-        hitbox_size = new_hitbox_data[1]
 
+    def apply_collision_detection(self, game_state):
+        hitbox_pos, hitbox_size = self.state[const.HITBOX_CONFIG].get(self.state[const.STATE])
         self.state[const.HITBOX] = Rect((self.state[const.X_COORD] + hitbox_pos[0], self.state[const.Y_COORD] + hitbox_pos[1]), hitbox_size)
 
         plats = [Rect((x, y), (w, h)) for x, y, w, h, idx in game_state[const.LEVEL][const.PLATFORMS]]
         brokeflag = self.state[const.HITBOX].collidelist(plats) != -1
 
+        # X axis
```

Comment:
In general do you mind leaving more descriptive comments? I totally appreciate that you know this game like the back of your hand, but for new devs joining the project it can be really helpful.

I personally still need to understand a lot more about the underlying game mechanics, so IMO it would provide immediate value add.